src/App.tsx

```diff
@@ -1,14 +1,73 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+  const [query, setQuery] = useState('');
+  const [status, setStatus] = useState('all');
+
+  useEffect(() => {
+    getTodos().then(setTodos);
+  }, []);
+
+  const visibleTodos = todos.filter(todo => {
+    const lowerTitle = todo.title.toLowerCase();
+    const normalizedQuery = query.toLowerCase().trim();
+
+    const isQueryMatch = lowerTitle.includes(normalizedQuery);
+    let isStatus = true;
+
+    switch (status) {
+      case 'active':
+        isStatus = !todo.completed;
+        break;
+
+      case 'completed':
+        isStatus = todo.completed;
+        break;
+
+      default:
+        isStatus = true;
+    }
+
+    return isQueryMatch && isStatus;
+  });
```

Comment:
i would suggest to use useMemo here, so that your todos will render only when the query, todos or status are changed.